airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/datetime_stream_slicer.py

```diff
@@ -37,22 +38,24 @@ def __init__(
         self._timezone = datetime.timezone.utc
         self._interpolation = JinjaInterpolation()
         self._datetime_format = datetime_format
-        self._start_time = self.parse_date(start_time.eval(config))
-        self._end_time = self.parse_date(end_time.eval(config))
-        self._end_time = min(self._end_time, datetime.datetime.now(tz=datetime.timezone.utc))
-        self._start_time = min(self._start_time, self._end_time)
+        self._start_date = start_date
+        self._end_date = end_date
```

Comment:
we are losing parse time validation when we move interpolation to runtime in `stream_slices()` unfortunately.